quinn-proto/src/frame.rs

```diff
@@ -929,8 +940,86 @@ impl AckFrequency {
     }
 }
 
+/* Address Discovery https://datatracker.ietf.org/doc/draft-seemann-quic-address-discovery/ */
+
+/// Conjuction of the information contained in the address discovery frames
+/// ([`FrameType::OBSERVED_IPV4_ADDR`], [`FrameType::OBSERVED_IPV6_ADDR`]).
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub(crate) struct ObservedAddr {
+    /// Monotonically increasing integer within the same connection.
+    pub(crate) seq_no: VarInt,
+    /// Reported observed address.
+    pub(crate) ip: IpAddr,
+    /// Reported observed port.
+    pub(crate) port: u16,
+}
+
+impl ObservedAddr {
+    pub(crate) fn new<N: Into<VarInt>>(remote: std::net::SocketAddr, seq_no: N) -> Self {
```

Comment:
Do we ever want to pass something that isn't already a `VarInt` in here? If not, skip the generic for simplicity.

Reply:
Also, import `SocketAddr` instead of qualifying it.